network-libp2p/src/network_types.rs

```diff
@@ -238,6 +239,8 @@ pub(crate) struct TaskState {
     pub(crate) dht_bootstrap_state: DhtBootStrapState,
     /// DHT (kad) is in server mode
     pub(crate) dht_server_mode: bool,
+    /// The NAT status of the local peer
+    pub(crate) nat_status: NatState,
```

Comment:
Maybe I'm missing something here, but why do we want this `nat_status`? I don't see any reader of this

Reply:
It is necessary to keep a state of the all the addresses in order to determine the final NAT state. With `autonatv1`, it would give you a NAT of `Public` or `Private` eventually but v2 only gives you the results of the probes. Thus you have to keep track of everything yourself.

Reply:
But who's reading this status? I am only seeing some logging related to this

Reply:
We discussed it internally and we decided to keep it for `Nat` status visibility.